rakan/cRakan/src/Runner.h

```diff
@@ -25,41 +27,51 @@ class Runner {
   /*
   * Default constructor.
   */
-  Runner() {}
+  Runner();
 
-  void Add();
+  Runner(uint32_t num_precincts, uint32_t num_districts);
 
+  void add_node(uint32_t node_id, uint32_t county, uint32_t minority, uint32_t majority);
+
+  void add_edge(uint32_t node_one, uint32_t node_two);
+
+  // Returns a vector of vectors of uint32_t, where each vector
+  // represents the changes made to the graph in each step of 
+  // the walk.
+  vector<vector<uint32_t>&>& getMaps();
   /*
-  * Sets the district assignments according to the given map.
-  * Map is interpreted as storing node ID as the key and district ID
-  * as the value.
+  * Sets the district assignments according to the given vector.
+  * Vector is interpreted as the node_id corresponding to the index
+  * of the district number inside of the vector.
   * 
-  * @param    map     The map to set the current graph's districts
-  *                   to be
+  * @param    districts     a vector of uint32_t where each index of
+  *                         the vector corresponds to the node id.
   * 
-  * @return SUCCESS if all assignment successful; the appropriate
-  *         error code otherwise
   */
-  uint16_t SetDistricts(unordered_map<uint32_t, uint32_t> &map);
+  void set_districts(vector<uint32_t>& districts);
 
   /*
   * Generates random seeds on the current graph. Randomly selects
   * a number of nodes to be the "center" of each district and assigns
   * other nodes reachable from the seed nodes to the respective
   * district.
   * 
-  * @return SUCCESS iff all seeding and assignment successful;
-  *         SEEDING_FAILED otherwise
   */
-  uint16_t SeedDistricts();
+  bool seed();
 
   /*
   * Populates the graph's data structures.
   * 
-  * @return SUCCESS iff all populating was successful; the appropriate
-  *         error code otherwise
   */
-  uint16_t PopulateGraphData();
+  void populate();
```

Comment:
Mark what methods this needs to be called ahead of time. (e.g. `seed`)

Reply:
It depends on the job you want done. If the job is to get the score of a graph, then `seed` does not need to be called before `populate`. Should we specify that?

Reply:
Yes please